src/components/expanded-state/ens/InfoRow.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import { useTheme } from '../../../context/ThemeContext';
+import { ShimmerAnimation } from '../../animations';
+import { Icon } from '../../icons';
+import {
+  Bleed,
+  Box,
+  Inline,
+  Inset,
+  Text,
+  useForegroundColor,
+} from '@rainbow-me/design-system';
+import { ImgixImage } from '@rainbow-me/images';
+
+export function InfoRowSkeleton() {
+  const { colors } = useTheme();
+  return (
+    <Inline alignHorizontal="justify" horizontalSpace="24px" wrap={false}>
+      <Box
+        style={{
+          backgroundColor: colors.alpha(colors.blueGreyDark, 0.04),
+          borderRadius: 12,
+          height: 24,
+          overflow: 'hidden',
+          width: 100,
+        }}
+      >
+        <ShimmerAnimation
+          color={colors.alpha(colors.blueGreyDark, 0.06)}
+          enabled
+          gradientColor={colors.alpha(colors.blueGreyDark, 0.06)}
+          width={100}
+        />
+      </Box>
+      <Box
+        style={{
+          backgroundColor: colors.alpha(colors.blueGreyDark, 0.04),
+          borderRadius: 12,
+          height: 24,
+          overflow: 'hidden',
+          width: 150,
+        }}
+      >
+        <ShimmerAnimation
+          color={colors.alpha(colors.blueGreyDark, 0.06)}
+          enabled
+          gradientColor={colors.alpha(colors.blueGreyDark, 0.06)}
+          width={150}
+        />
+      </Box>
+    </Inline>
+  );
+}
+
+export default function InfoRow({
+  icon = undefined,
+  isImage = false,
+  label,
+  wrapValue = children => children,
+  value = undefined,
+  useAccentColor,
+}: {
+  icon?: string;
+  isImage?: boolean;
+  label: string;
+  wrapValue?: (children: React.ReactNode) => React.ReactNode;
+  value?: string;
+  useAccentColor?: boolean;
+}) {
+  const { colors } = useTheme();
+  const accentColor = useForegroundColor('accent');
+
+  const [show, setShow] = useState(isImage);
+  const [isMultiline, setIsMultiline] = useState(false);
+
+  return (
+    <Inline alignHorizontal="justify" horizontalSpace="24px" wrap={false}>
+      <Box style={{ minWidth: 60, opacity: show ? 1 : 0 }}>
+        <Inset top={isMultiline ? '15px' : '10px'}>
+          <Text color="secondary60" size="16px" weight="bold">
+            {label}
+          </Text>
+        </Inset>
+      </Box>
+      {wrapValue(
+        <>
+          {isImage ? (
+            <Box
+              as={ImgixImage}
+              borderRadius={16}
+              flexShrink={1}
+              height="64px"
+              source={{ uri: value }}
+              width="full"
+            />
+          ) : (
+            <Box
+              borderRadius={16}
+              flexShrink={1}
+              onLayout={({
+                nativeEvent: {
+                  layout: { height },
+                },
+              }) => {
+                setIsMultiline(height > 40);
+                setShow(true);
```

Comment:
maybe moving this method out for easier readability?

Reply:
sounds good!